snuba/snapshots/postgres_snapshot.py

```diff
@@ -120,12 +121,44 @@ def load(cls, product: str, path: str) -> PostgresSnapshot:
     def get_descriptor(self) -> PostgresSnapshotDescriptor:
         return self.__descriptor
 
+    def __table_iterable(self, csv_table) -> Generator[Mapping[str, Any], None, None]:
+        for r in csv_table:
+            yield r
+
     @contextmanager
-    def get_table_file(self, table: str) -> Generator[IO[bytes], None, None]:
+    def get_table_file(
+        self,
+        table: str,
+    ) -> Generator[Iterable[Mapping[str, Any]], None, None]:
         table_path = os.path.join(self.__path, "tables", "%s.csv" % table)
         try:
-            with open(table_path, "rb") as table_file:
-                yield table_file
+            with open(table_path, "r") as table_file:
+                csv_file = csv.DictReader(table_file)
```

Comment:
I am still playing with csv reader parameters to be able to load null values and escape value properly.